publication-model/src/main/java/no/unit/nva/api/PublicationResponse.java

```diff
@@ -99,15 +98,15 @@ public static PublicationResponse fromPublication(Publication publication) {
         return response;
     }
 
-    private static AssociatedArtifactList getPublicAssociatedArtifacts(Publication publication) {
-        var artifacts = publication.getAssociatedArtifacts().stream()
-                            .filter(PublicationResponse::isPublicAssociatedArtifact)
-                            .toList();
-        return new AssociatedArtifactList(artifacts);
+    public void setAssociatedArtifacts(List<AssociatedArtifactResponse> associatedArtifactResponses) {
+        this.associatedArtifacts = associatedArtifactResponses;
     }
 
-    private static boolean isPublicAssociatedArtifact(AssociatedArtifact artifact) {
-        return artifact instanceof PublicAssociatedArtifact;
+    private static List<AssociatedArtifactResponse> getPublicAssociatedArtifacts(Publication publication) {
+        return publication.getAssociatedArtifacts().stream()
+                   .filter(a -> AssociatedArtifact.PUBLIC_ARTIFACT_TYPES.contains(a.getClass()))
```

Comment:
artifact

Reply:
probably extract `isPublicArtifact` although we call them `open`?